src/nrniv/shape.cpp

```diff
@@ -1040,24 +1036,17 @@ PolyGlyph* ShapeScene::shape_section_list() {
 }
 
 void ShapeScene::name(const char* s) {
-    if (!var_name_) {
-        var_name_ = new CopyString(s);
-    } else {
-        *var_name_ = s;
-    }
+    var_name_ = s;
 }
 
 void ShapeScene::save_phase2(std::ostream& o) {
-    char buf[256];
-    if (var_name_) {
-        if ((var_name_->string())[var_name_->length() - 1] == '.') {
-            Sprintf(buf, "%sappend(save_window_)", var_name_->string());
+    if (!var_name_.empty()) {
+        if (var_name_.back() == '.') {
+            o << var_name_ << "append(save_window_)" << std::endl;
         } else {
-            Sprintf(buf, "%s = save_window_", var_name_->string());
+            o << var_name_ << " = save_window_" << std::endl;
         }
-        o << buf << std::endl;
-        Sprintf(buf, "save_window_.save_name(\"%s\")", var_name_->string());
-        o << buf << std::endl;
+        o << "save_window_.save_name(\"" << var_name_ << "\")" << std::endl;
```

Comment:
just a comment: these are straightforward but having test somewhere about this would have been nice!